api-resources.md

```diff
@@ -0,0 +1,414 @@
+
+If the collection is available, the response body is a `struct Collection`. Otherwise the server can return something like HTTP 202 Accepted to indicate it's not ready yet, or HTTP 404 if no such aggregate share request is known to the server.
+
+This is analogous to DAP-02's `GET` on a collect job URI.
+
+##### PUT
+
+PUT to a collection resource is how the collector instructs the leader to assemble the aggregate shares. The request is an idempotent PUT, because if a collection with the provided ID already exists, its representation can be returned by the helper without further state mutation.
+
+The request body is a `CollectReq`:
+
+```
+struct {
+  QueryType query_type;
+  select (Query.query_type) {
+    case time_interval: Interval batch_interval;
+    case fixed_size: Empty;
+  }
+} Query;
+
+struct {
+  Query query;
+  opaque agg_param<0..2^16-1>; /* VDAF aggregation parameter */
+} CollectReq;
+```
+
+If successful, the response from the helper is a `struct Collection`. That same share may later be obtained by a GET request on the resource.
+
+This is analogous to DAP-02's `POST /collect`.
+
+###### A note on `Query.fixed_size`
+
+`struct Query` is essentially identical to `struct BatchSelector`, so the note about `BatchSelector.fixed_size` from `PUT aggregate_share` applies here, except that a collection's ID is chosen by the collector.
+
+##### DELETE
+
+Instructs the leader to abandon the collection and allows the leader to discard all state related to it. This is akin to DELETE on an aggregate share for the helper (indeed, the leader might send a DELETE to the helper's aggregate share resource after its corresponding collection is deleted).
+
+This is analogous to DAP-02's `DELETE` on a collect job URI.
+
+#### Alternatives: `collection-id`, `PUT` vs. `POST` and idempotence?
+
+This proposal introduces the notion of a `collection-id`, which is chosen by the collector when it sends `CollectReq` to helper. In DAP-02, a collection is already uniquely identified by the combination of the aggregation parameter and the query, meaning we can get by without `collection-id`. Let's sketch out what the API would look like so we can weigh pros and cons.
+
+`POST /tasks/task-id/collections`
+
+The body is a `CollectReq`, defined as above. The request method is `POST` instead of `PUT` because I don't think this request can be idempotent. Let's say a collector makes a time interval query over some interval _i_ and it is received by the leader at time _t1_. At _t1_, the leader has _n1_ reports that fall into the interval _i_. Then, suppose at least one more report that falls into _i_ arrives and the leader and helper prepare it. Then, at _t2 > t1_, the collector sends a query with the same interval _i_ again.
+
+Should the leader serve up the results it computed at time _t1_? Or should it make a new collection, consuming another unit of max batch query, that includes the reports that arrived and were prepared between _t1_ and _t2_?
```

Comment:
I'm confused, wouldn't the leader ignore the reports that fall into interval _i_ after the first collection request at _t1_? 

From DAP-02: "The Leader MUST ignore any report pertaining to a batch that has already been collected."

Multiple collections for a single batch are to allow aggregation parameter to vary; I don't think the current wording of DAP-02 allows for collecting "the same batch but with a few more reports that have arrived since the last collection" to avoid potentially breaking privacy of those new reports.

Reply:
You are correct, I forgot about that requirement in DAP-02. I need to revisit this text after thinking harder about `current-batch` and fixed batch queries in general.